src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -256,9 +270,12 @@ protected function render(string $view, array $parameters = [], Response $respon
             $response = new Response();
         }
 
-        if (200 === $response->getStatusCode()) {
+        if ($this->renderFormViews && 200 === $response->getStatusCode()) {
             foreach ($parameters as $v) {
-                if ($v instanceof FormInterface && $v->isSubmitted() && !$v->isValid()) {
+                if (
+                    $v instanceof FormInterface && $v->isSubmitted() && !$v->isValid()
+                    || $v instanceof FormView && (!$v->vars['valid'] ?? true)
```

Comment:
shouldn't this be `!($v->vars['valid'] ?? true)` instead ?

Reply:
Indeed, fixed. Thanks!